dev/NavigationView/NavigationView.cpp

```diff
@@ -1930,13 +1936,32 @@ void NavigationView::ChangeSelection(const winrt::IInspectable& prevItem, const
         UnselectPrevItem(prevItem, nextItem);
         ChangeSelectStatusForItem(nextItem, true /*selected*/);
 
+        // Selection changed and we need to notify UIA
+        // HOWEVER expand collapse can also trigger if an item can expand/collapse
+        // There are multiple cases when selectino changes:
+        // - Through click on item with no children -> No expand/collapse change
+        // - Through click on item with children -> Expand/collapse change
+        // - Through API with item without children -> No expand/collapse change
+        // - Through API with item with children -> No expand/collapse change
+        if (!m_shouldIgnoreUIASelectionRaiseAsExpandCollapseWillRaise)
+        {
+            if (winrt::AutomationPeer peer = winrt::FrameworkElementAutomationPeer::FromElement(*this))
+            {
+                auto navViewItemPeer = peer.as<winrt::NavigationViewAutomationPeer>();
+                winrt::get_self<NavigationViewAutomationPeer>(navViewItemPeer)->RaiseSelectionChangedEvent(
+                    prevItem, nextItem
+                );
+            }
+        }
+        m_shouldIgnoreUIASelectionRaiseAsExpandCollapseWillRaise = false;
```

Comment:
I think  we should GSL scope guard this.

Reply:
Fixed